scipy/spatial/transform/rotation.py

```diff
@@ -1775,10 +1775,73 @@ def random(cls, num=None, random_state=None):
+    @np.deprecate(message="match_vectors is deprecated in favor of "
+                          "align_vectors in scipy 1.4.0 and will be removed "
+                          "in scipy 1.6.0")
+    def match_vectors(cls, a, b, weights=None, normalized=False):
+        """Deprecated in favor of `align_vectors`."""
+        a = np.asarray(a)
+        if a.ndim != 2 or a.shape[-1] != 3:
+            raise ValueError("Expected input `a` to have shape (N, 3), "
+                             "got {}".format(a.shape))
+        b = np.asarray(b)
+        if b.ndim != 2 or b.shape[-1] != 3:
+            raise ValueError("Expected input `b` to have shape (N, 3), "
+                             "got {}.".format(b.shape))
+
+        if a.shape != b.shape:
+            raise ValueError("Expected inputs `a` and `b` to have same shapes"
+                             ", got {} and {} respectively.".format(
+                                a.shape, b.shape))
+
+        if b.shape[0] == 1:
+            raise ValueError("Rotation cannot be estimated using a single "
+                             "vector.")
+
+        if weights is None:
+            weights = np.ones(b.shape[0])
+        else:
+            weights = np.asarray(weights)
+            if weights.ndim != 1:
+                raise ValueError("Expected `weights` to be 1 dimensional, got "
+                                 "shape {}.".format(weights.shape))
+            if weights.shape[0] != b.shape[0]:
+                raise ValueError("Expected `weights` to have number of values "
+                                 "equal to number of input vectors, got "
+                                 "{} values and {} vectors.".format(
+                                    weights.shape[0], b.shape[0]))
+        weights = weights / np.sum(weights)
+
+        if not normalized:
+            a = a / scipy.linalg.norm(a, axis=1)[:, None]
+            b = b / scipy.linalg.norm(b, axis=1)[:, None]
+
+        B = np.einsum('ji,jk->ik', weights[:, None] * a, b)
+        u, s, vh = np.linalg.svd(B)
+
+        # Correct improper rotation if necessary (as in Kabsch algorithm)
+        if np.linalg.det(u @ vh) < 0:
+            s[-1] = -s[-1]
+            u[:, -1] = -u[:, -1]
+
+        C = np.dot(u, vh)
+
+        zeta = (s[0]+s[1]) * (s[1]+s[2]) * (s[2]+s[0])
+        if np.abs(zeta) <= 1e-16:
+            raise ValueError("Three component error vector has infinite "
+                             "covariance. It is impossible to determine the "
+                             "rotation uniquely.")
+
+        kappa = s[0]*s[1] + s[1]*s[2] + s[2]*s[0]
+        sensitivity = ((kappa * np.eye(3) + np.dot(B, B.T)) /
+                       (zeta * a.shape[0]))
+        return cls.from_matrix(C), sensitivity
 
-        Find a rotation between frames A and B which best matches a set of
+    @classmethod
+    def align_vectors(cls, a, b, weights=None, return_sensitivity=False):
```

Comment:
Much of the error handling code in this new function is not being hit by tests according to the Azure cov report

Reply:
I'll add tests for it.

Reply:
Verified, thanks.